### PR TITLE
Added packagePattern option to config

### LINKED ISSUE #327: PackagePattern in naming conventions
I posted a question in StackOverflow (https://stackoverflow.com/questions/45846555/detekt-snake-case-package) because I can't specify the naming conventions for packages names.

Is it possible?

### PR DESCRIPTION
**Pull request by schalkms**
I also spotted that we have two tests for the `NamingConventionViolation` rule. I think one should be deleted because they are basically testing the same thing.

EDIT: fixes #327

### CODE REVIEW

**Comment by Mauin** [comment]
This shows that we need to find a way (Wiki, `.md` files in the codebase somewhere or something) to document rules and their configuration better. Currently our `default-detekt-config.yml` is pretty much all the documentation we have and not all parameters of a rule are usually added there.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/NamingConventionSpec.kt [hunk 1]
```diff
@@ -1,62 +0,0 @@
...
-import io.gitlab.arturbosch.detekt.test.compileContentForTest
-import io.gitlab.arturbosch.detekt.test.lint
-import org.assertj.core.api.Assertions
-import org.jetbrains.spek.api.Spek
-import org.jetbrains.spek.api.dsl.given
-import org.jetbrains.spek.api.dsl.it
-
-class NamingConventionSpec : Spek({
-
-	given("a kt file with crazy naming schemes") {
```
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/NamingConventionSpec.kt]: Should we migrate these test cases over to the other test class to keep the same coverage?

Comment (schalkms) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/NamingConventionSpec.kt]: I'm working on it. I don't know what problem the appveyor build had with my first commit.

Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/NamingConventionSpec.kt]: I also noticed the appveyor problem in some of my commits. Kotlin.Unit cannot be loaded ^^

**Comment by arturbosch** [comment]
It looks rdy to merge?

**Comment by schalkms** [comment]
@arturbosch yes

**Pull request merged**